### ISSUE TITLE
SGPR build likelihood is incorrect for Multioutput

### ISSUE THREAD

**Issue by sanket-kamthe** [GPflow/GPflow#831]
The multiplication by output_dim in the likelihood is not needed if using multi-output kernels
link (https://github.com/GPflow/GPflow/blob/develop/gpflow/models/sgpr.py#L175)

See line 175 below: 

```
        bound += -0.5 * output_dim * tf.reduce_sum(Kdiag) / self.likelihood.variance
        bound += 0.5 * output_dim * tf.reduce_sum(tf.linalg.diag_part(AAT))
```
This is correct if `Y` is multidim for single kernel. 

There is no clean way I could think of handling this issue without a discussion. 

Also, should we have had a test to find this bug ?

**Comment by hughsalimbeni** [comment]
How about something like `output_dim = tf.where(tf.less(tf.rank(Kdiag), 2), output_dim, 1)`?

**Comment by sanket-kamthe** [comment]
Well, I would like to avoid putting conditions on the gradient path of the model, especially when we can determine this before compiling the graph. 

Are we anticipating a scenario where `rank(kdiag) ` changes at run time ?

**Comment by vdutor** [comment]
I haven't tested this my self, but I think you can have a condition based on `len(Kdiag.shape) == 2` which should work at graph compiling time.

**Comment by hughsalimbeni** [comment]
I'd suggest that `len(Kdiag.shape) == 2` can be dangerous if the shape is None, as at run time the shape could then be anything

**Comment by st--** [comment]
@sanket-kamthe it's an old issue but still seems relevant - would be great if you'd be up for writing a test (that currently would fail, but that we can work towards fixing). Thank you!

**Comment by st--** [comment]
Closing this in favour of the larger issue captured in #1209
